src/main/java/co/rsk/bitcoinj/core/CheckpointManager.java

```diff
@@ -139,6 +144,17 @@ private Sha256Hash readBinary(InputStream inputStream) throws IOException {
                 buffer.position(0);
                 checkpoints.put(block.getHeader().getTimeSeconds(), block);
             }
+
+            int actualCheckpointsSize = dis.available();
+            int expectedCheckpointsSize = numCheckpoints * size;
+            // Check if there are any bytes left in the stream. If it does, it means that checkpoints are malformed
+            if (dis.available() > 0) {
```

Comment:
```suggestion
            if (actualCheckpointsSize > 0) {
```
You already stored that value in a variable, right?

Reply:
Updated